submissions/nadsatt/post-oop/index.js

```diff
@@ -0,0 +1,152 @@
+/* 
+   Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/nadsatt/a-tiny-JS-world
+   Web app: https://nadsatt.github.io/a-tiny-JS-world/
+*/
+
+// ======== OBJECTS DEFINITIONS ========
+
+/* WHAT WAS DONE to make my code SOLID:
+   - FOR S: i created new methods 'addFriends' and 'getFriends' in Inhabitant class and moved it to separate class 'FriendsManager'
+            i also wanted to move 'toString' method to separate class 'InhabitantConverter 
+            but i'm not sure about both ideas because it makes my code strange a bit (especially when i need to create FriendsManager instance for each inhabitant)
+   - FOR O: i changed 'toString' method in 'Inhabitant' to make it dependent on specific properties 
+            to have ability to redefine this method in subclasses
+            to show how this method can be simultaneously closed for modifications in superclass and open for modifications in subclasses
+   - FOR L: as i see, there is no cases where subclass can't be used instead of its superclass
+            for example, in 'addFriends' method of 'FriendsManager' class instance of any subclass can be used
+   - FOR I: as i see, there is no unused/unrelevant superclass' methods/props in subclasses
+   - FOR D: -
+*/ 
+
+class Inhabitant {
+   constructor(species, saying, gender, name, friends = []){
+      this.species = species;
+      this.saying = saying;
+      this.gender = gender;
+      this.name = name;
+      this.friends = friends;
+   }
+
+   toString(){
+      return `species: ${this.species}, saying: ${this.saying}, gender: ${this.gender}, name: ${this.name}, friends: ${new FriendsManager(this).getFriends()}`;
+   }
+}
+
+class FriendsManager {
+   constructor(inhabitant){
```

Comment:
**FIX REQUIRED**

Once we have FriendsManager then we could instantiate it in the base class constructor and keep it as a property of the base class. `class Inhabitant{contsructor(...) { this.friendsManager = new FriendsManager(..)...`
This allows you:
- initialize friends via friends manager `constructor(inhabitant, friends = [])`
- you do not need to interact with `this.friends` in base class at all, even in constructor; yet base class will still have the property
- you will refer to `this.friendsManager` in `Inhabitant.toString` instead of instantiation